src/k8s-extension/azext_k8s_extension/partner_extensions/WorkloadIAM.py

```diff
@@ -0,0 +1,139 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import subprocess
+
+from knack.log import get_logger
+from knack.util import CLIError
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+
+from ..vendored_sdks.models import (Extension, Scope, ScopeCluster)
+
+from .DefaultExtension import DefaultExtension
+
+logger = get_logger(__name__)
+
+CONFIG_SETTINGS_USER_TRUST_DOMAIN = 'trustDomain'
+CONFIG_SETTINGS_USER_LOCAL_AUTHORITY = 'localAuthority'
+CONFIG_SETTINGS_USER_TENANT_ID = 'tenantID'
+CONFIG_SETTINGS_USER_JOIN_TOKEN = 'joinToken'
+
+CONFIG_SETTINGS_HELM_TRUST_DOMAIN = 'global.workload-iam.trustDomain'
+CONFIG_SETTINGS_HELM_TENANT_ID = 'global.workload-iam.tenantID'
+CONFIG_SETTINGS_HELM_JOIN_TOKEN = 'workload-iam-local-authority.localAuthorityArgs.joinToken'
+
+class WorkloadIAM(DefaultExtension):
+
+    def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, cluster_rp,
+               extension_type, scope, auto_upgrade_minor_version, release_train, version, target_namespace,
+               release_namespace, configuration_settings, configuration_protected_settings,
+               configuration_settings_file, configuration_protected_settings_file,
+               plan_name, plan_publisher, plan_product):
+        """
+        Create method for ExtensionType 'microsoft.workloadiam'.
+        """
+
+        # Ensure that the values provided by the user for generic values of Arc extensions are
+        # valid, set sensible default values if not.
+        if release_train is None:
+            # TODO - Set this to 'stable' when the extension is ready
+            release_train = 'preview'
+
+        if scope is None:
+            scope = 'cluster'
+        elif scope != 'cluster':
+            raise InvalidArgumentValueError(
+                f"Invalid scope '{scope}'. This extension can only be installed at 'cluster' scope.")
+
+        # Scope is always cluster
+        scope_cluster = ScopeCluster(release_namespace=release_namespace)
+        ext_scope = Scope(cluster=scope_cluster, namespace=None)
+
+        # Get user configuration values and remove them from the dictionary
+        trust_domain = configuration_settings.pop(CONFIG_SETTINGS_USER_TRUST_DOMAIN, None)
```

Comment:
>trust_domain = configuration_settings.pop(CONFIG_SETTINGS_USER_TRUST_DOMAIN, None) [](http://example.com/codeflow?start=8&length=82)

Is your intent to get the value for this key?  If so, since configuration_settings is a dictionary, don't you want to use get?  AFAIK, string.pop removes the input value, which in this case is the key - so you will end up having the key in the variable 'trust_domain', no?
Have you tested this to confirm this works?

Reply:
Ah!  Got it - thanks for explaining!  It wasn't clear from the comment in the code.  However, as long as it makes sense to you, it's good :)